code/datums/ship.dm

```diff
@@ -90,6 +94,12 @@ GLOBAL_VAR(next_ship_id)
 	for(var/i in init_components)
 		var/datum/component/component = SSship.cname2component(init_components[i])
 		var/datum/component/C = new component.type
+		if(istype(C, /datum/component/weapon/carrier_weapon) && (ship_boarders || num_boarders)) // we have a carrier weapon and something to change on it
+			var/datum/ship_attack/carrier_weapon/A = C.attack_data
+			if(ship_boarders)
+				A.boarding_mob = ship_boarders
+			if(num_boarders)
+				A.amount = num_boarders
```

Comment:
hi, my name is idiot who didn't know that weapons had a reference to the ship they are bolted to